File: app-python/entitlements_app/requests/add_feature_to_plan_request.py
"""Pydantic request model for command `AddFeatureToPlan`."""

from pydantic import BaseModel


class AddFeatureToPlanRequest(BaseModel):
    allowance: int
    effectiveAt: str
    enforcementMode: str
    entitlementId: str
    featureKey: str
    planId: str
    tenantId: str
    unit: str

    @property
    def effective_at(self) -> str:
        return self.effectiveAt

    @property
    def enforcement_mode(self) -> str:
        return self.enforcementMode

    @property
    def entitlement_id(self) -> str:
        return self.entitlementId

    @property
    def feature_key(self) -> str:
        return self.featureKey

    @property
    def plan_id(self) -> str:
        return self.planId

    @property
    def tenant_id(self) -> str:
        return self.tenantId

File: app/src/main/java/com/example/spring/AddFeatureToPlanRequest.java
package com.example.spring;

/**
 * Request DTO for command AddFeatureToPlan.
 */
public record AddFeatureToPlanRequest(
    int allowance,
    @jakarta.validation.constraints.NotBlank String effectiveAt,
    @jakarta.validation.constraints.NotBlank String enforcementMode,
    @jakarta.validation.constraints.NotBlank String entitlementId,
    @jakarta.validation.constraints.NotBlank String featureKey,
    @jakarta.validation.constraints.NotBlank String planId,
    @jakarta.validation.constraints.NotBlank String tenantId,
    @jakarta.validation.constraints.NotBlank String unit
) {}
